### PR TITLE
Add async capabilities to react select

### PR DESCRIPTION
**Pull request by ousmaneo**
Before these changes, Select with a high number of options (1000) would result in performance problems. This #12250 from @maxiadlovskii adds the ability to use react-window to virtualize the list when the number of options is high, giving better performance.
With this change, we are adding a layer that is using [react-window-infinite-loader](https://github.com/bvaughn/react-window-infinite-loader) to be able to load items asynchronously when scrolling. This also add the ability to filter the options using an async call.

As an example, we added a new UserSelectField that can be used in place of a users list select.

## Motivation and Context

For context see [issue](https://github.com/Graylog2/graylog2-server/issues/12453) and [task](https://github.com/Graylog2/graylog2-server/issues/12819)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Comment by gally47** [comment]
Screenshot 2022-07-25 at 14 29 05

It will be good to adjust the height of the menu to the number of users that we can avoid having the extra useless space.

**Comment by mnin** [comment]
I took another look at this PR. When I do a search, even with very few users (under 20) I get the same user twice in the list:

Screenshot 2022-08-24 at 14 41 12

**Review by mnin — APPROVED** [review]
LGTM

**Review by gally47 — APPROVED** [review]

File: graylog2-web-interface/src/components/common/Select/CustomMenuList.tsx [hunk 1]
```diff
@@ -51,28 +51,30 @@ const Row = ({ data, index, setSize, style, containerWidth }: RowProps) => {
   );
 };
 
-const WindowList = ({ children }: Props.MenuList) => {
+export const WindowList = ({ children, listRef, ...rest }: Props.MenuList & { listRef?: any}) => {
   const containerRef = useRef(null);
-  const listRef = useRef(null);
+  const vListRef = useRef(null);
```
[the remark this answers is not in the document]
Comment (ousmaneo) [reply graylog2-web-interface/src/components/common/Select/CustomMenuList.tsx:56]: Thank you for having a look, the `vListRef` is used as ref when the `listRef` is indefined or null.

File: graylog2-web-interface/src/components/users/UsersSelectField.tsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    onChange: (nextValue) => string,
+}
+
+const UsersSelectField = ({ value, onChange }: Props) => {
+  const currentUser = useContext(CurrentUserContext);
+  const [paginatedUsers, setPaginatedUsers] = useState<PaginatedUsers | undefined>();
+  const loadUsersPaginated = useCallback((pagination = DEFAULT_PAGINATION) => {
+    if (isPermitted(currentUser.permissions, 'users:list')) {
+      return UsersDomain.loadUsersPaginated(pagination).then((newPaginatedUser) => {
+        return newPaginatedUser;
```
[the remark this answers is not in the document]
Comment (ousmaneo) [reply graylog2-web-interface/src/components/users/UsersSelectField.tsx:52]: The then part was introduced to be able to set the loading state.

**Pull request merged**